aruco_opencv/src/aruco_tracker.cpp

```diff
@@ -171,9 +180,15 @@ class ArucoTracker : public rclcpp_lifecycle::LifecycleNode
 
     auto qos = rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(image_sub_qos), image_sub_qos);
 
-    img_sub_ = create_subscription<sensor_msgs::msg::Image>(
-      cam_base_topic_, qos, std::bind(
-        &ArucoTracker::callback_image, this, std::placeholders::_1));
+    if (image_sub_compressed_) {
+      compressed_img_sub_ = create_subscription<sensor_msgs::msg::CompressedImage>(
+        cam_base_topic_ + "/compressed", qos, std::bind(
+          &ArucoTracker::callback_compressed_image, this, std::placeholders::_1));
+    } else {
+      img_sub_ = create_subscription<sensor_msgs::msg::Image>(
+        cam_base_topic_, qos, std::bind(
+          &ArucoTracker::callback_image, this, std::placeholders::_1));
+    }
```

Comment:
Assuming the previous suggestion, we can now use the fully resolves image topic name to create the compressed image topic name.
```suggestion
    if (image_sub_compressed_) {
      compressed_img_sub_ = create_subscription<sensor_msgs::msg::CompressedImage>(
        image_topic + "/compressed", qos, std::bind(
          &ArucoTracker::callback_compressed_image, this, std::placeholders::_1));
    } else {
      img_sub_ = create_subscription<sensor_msgs::msg::Image>(
        image_topic, qos, std::bind(
          &ArucoTracker::callback_image, this, std::placeholders::_1));
    }
```